src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.cpp

```diff
@@ -139,21 +139,49 @@ Value insertAllocAndDeallocSimple(PatternRewriter &rewriter, Operation *op,
 }
 
 // Determine if current function returns the result value of the
-// current op being lowered. If it does then dealloc should not be
-// inserted.
+// current op or the result value of reinterpret_cast op whose
+// operand is the result value of current op. If it does then
+// dealloc should not be inserted.
 bool checkInsertDealloc(Operation *currentOp, int resultIndex) {
   auto parentBlock = currentOp->getBlock();
-
   bool insertDealloc = true;
-  parentBlock->walk([&insertDealloc, currentOp, resultIndex](ReturnOp op) {
-    // If there is at least one result to investigate.
-    if (currentOp->getNumResults() > 0) {
```

Comment:
Commenting here in the old code, since this code was used to generate the new version.

Isn't this test `currentOp->getNumResults() > 0` invariant to the walk? Meaning the currentOp is constant. If so, why is it not around the entire walk...
```
if (currentOp->getNumResults() > 0) {
  parentBlock->Walk(...)
}
```

Reply:
Second comment: Isn't it true that some operations can have multiple results? Is it enough to focus only on the first result here? Again, I understand that you probably did not write this code originally, but reading it now make me wonder if it is correct.

Reply:
@AlexandreEichenberger Thanks for your comments! I reflected the first comments.
Regarding second comment,  ReinterpretCastOp, ONNXReshapeOp, ONNXSqueezeV11Op, and ONNXUnsqueezeV11Op are processed here. Since they have only one result, I get only the first result.